src/config/base.cpp

```diff
@@ -203,8 +204,8 @@ std::vector<std::string> ConfigBase::_merge(
     //   might be our current config, or might be one single one of the new incoming messages).
     // - confs that failed to parse (we can't understand them, so leave them behind as they may be
     //   some future message).
-    int superconf = new_conf->unmerged_index();  // -1 if we had to merge
-    for (int i = 0; i < all_hashes.size(); i++) {
+    size_t superconf = new_conf->unmerged_index();  // -1 if we had to merge
```

Comment:
This is potentially problematic: `superconf` here uses -1 as a special had-to-merge value, but as a size_t that becomes a large positive integer.  I think we want to fix the signed-compare-mismatch here the other way, by using a `static_cast<int>(all_hashes.size())`.

Reply:
Actually this should probably be an optional<size_t> instead; I'm redoing it as such.